homework/index.js

```diff
@@ -30,18 +30,70 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
+  function createLayout() {
+    const root = document.getElementById('root');
+    createAndAppend('select', root, { id: 'repo-select' });
+    createAndAppend('div', root, { id: 'body-container' });
+    createAndAppend('div', document.getElementById('body-container'), { id: 'repo-details' });
+    createAndAppend('div', document.getElementById('body-container'), { id: 'contributors' });
+
+  }
+
+  function getRepoDataFromOrgAndAddToDOM() {
+    const REPOS_URL = 'https://api.github.com/orgs/foocoding/repos?per_page=100';
+    fetchJSON(REPOS_URL, (err, arrayOfRepoData) => { 
       if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+        alert(err.message);
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        arrayOfRepoData.sort((a, b) => (a.name.toLowerCase() > b.name.toLowerCase()) ? 1 : -1);
+        arrayOfRepoData.forEach(repoDataObj => {
+          createAndAppend('option', document.getElementById('repo-select'), { text: repoDataObj.name })
+        })
+        addListenerOnSelect(arrayOfRepoData);
       }
-    });
+    })    
   }
 
-  const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
+  function addDataToRepoDetails(data) {
+    const repoDetailsDiv = document.getElementById('repo-details');
+    repoDetailsDiv.innerHTML = '';
+    createAndAppend('div', repoDetailsDiv, { id: "repoNameID" });
+    createAndAppend('a', document.getElementById("repoNameID"), {href: data.html_url, target: "_blank", text: data.name})
+    createAndAppend('div', repoDetailsDiv, { text: `Repository Description: ${data.description}` });
+    createAndAppend('div', repoDetailsDiv, { text: `Number of Forks: ${data.forks}` });
+  }
+
+  function getContributors(data) {
+    const contributorsDiv = document.getElementById('contributors');
+    contributorsDiv.innerHTML = 'Contributors:';
+    fetchJSON(data.contributors_url, (err, arrayOfContributorData) => {
+      if (err) {
+        alert(err.message);
+      } else {
+        arrayOfContributorData.forEach(repoObj => {
```

Comment:
Overall, great naming of functions and variables!

![Nice name](https://media.giphy.com/media/sVgI6O8HiAixq/giphy.gif)